docs/platforms/java/common/enriching-events/scopes/index.mdx

```diff
@@ -1,46 +1,109 @@
-discovers the current hub and asks it to capture an event. Internally the hub will
-then merge the event with the topmost scope's data.
+### Isolation Scope
+
+The isolation scope is used to isolate events from each other. For example, each request in a web server might get its own isolation scope, so that events from one request don't interfere with events from another request. In most cases, you'll want to put data that should be applied to your events on the isolation scope - which is also why all `Sentry.setXXX` methods, like `Sentry.setTag()`, will write data onto the currently active isolation scope. A classic example for data that belongs on the isolation scope is a user - each request may have a different user, so you want to make sure that the user is set on the isolation scope.
+
+You can modify the isolation scope via `Sentry.configureScope(ScopeType.ISOLATION, isolationScope -> { ... })`, but usually you'll just use the `Sentry.setXXX` methods to set data on the currently active isolation scope:
+
+```java
+Sentry.setTag("my-tag", "my value");
+// Is identical to:
+Sentry.configureScope(ScopeType.ISOLATION, scope -> {
+  scope.setTag("my-tag", "my value");
+});
+```
+
+### Current Scope
+
+The current scope is the local scope that is currently active. Unlike the rarely-forked isolation scope, the current scope may be forked more frequently and under the hood. It can be used to store data that should only be applied to specific events. In most cases, you should not access this scope directly, but use `Sentry.withScope` to create a new scope that is only active for a specific part of your code:
+
+```java
+Sentry.withScope(scope -> {
+  // scope is the current scope inside of this callback!
+  scope.setTag("my-tag", "my value");
+  // this tag will only be applied to events captured inside of this callback
+  // the following event will have the tag:
+  Sentry.captureException(new Exception("my error"));
+});
+// this event will not have the tag:
+Sentry.captureException(new Exception("my other error"));
+```
+
+You can modify the current scope via `Sentry.configureScope(ScopeType.CURRENT, scope -> { ... })`, but usually you should use `Sentry.withScope()` to interact with local scopes instead.
+
+## How Scope Data is Applied to Events
+
+Global scope, isolation scope and current scope are combined before an event (like an error or transaction) is sent to Sentry.
+In most cases, setting something on current scope replaces the same thing that may have been set on isolation or global scope. If it hasn't been set on current scope, the value on isolation scope takes precedence over the one from global scope. If there also isn't any value on isolation scope, the one from global scope is used if present.
+
+Note, there are exceptions to this, where values from all scopes are merged. This is the case for breadcrumbs, tags, extras, contexts, attachments and event processors.
+
+```java
```

Comment:
Should we also add a Kotlin sample here?

Reply:
Yeah, let's do it in a follow up PR, I've created https://github.com/getsentry/sentry-java/issues/4067 so we don't forget.